class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,111 @@
+        file.close();
+    }
+    void SaveData() {
+        std::ofstream file(databasePath, std::ios::trunc);
+        for (const auto& [name, count] : statsData ) {
+            file << name << " " << count << std::endl;
         }
-        fileEdit << WordToFind << " 1" << std::endl;
-        fileEdit.close();
+        file.close();
     }
-    else {
-        if (argument2 == "delete") {
+};
 
-            for (int i = 0; i < lines.size(); i++)
-            {
-                if (splitString(lines[i], ' ')[0] == WordToFind)
-                {
-                    lines.erase(lines.begin() + i);
-                }
-            }
-            std::ofstream fileEdit("database.txt", std::ios::trunc);
-            for (const auto& line : lines) {
-                if (line != "")
-                {
-                    fileEdit << line << std::endl;
-                }
-            }
-            std::cout << "All data about this user was cleared" << std::endl;
+int notFoundNameHandling(const std::string& additionalCommand, const std::string& nameToFind, Database& db ){
+    if (additionalCommand == deleteCommand) {
+        std::println("There is no such user ");
+        return 1;
+    }
+    db.UserIncrement(nameToFind);
+    std::println("Welcome {}",nameToFind);
+    return 0;
+}
+
+int FoundNameHandling(const std::string& additionalCommand, const std::string& nameToFind, Database& db ){
+    if (additionalCommand == deleteCommand) {
+        if (db.deleteUser(nameToFind)==true){
+            std::println("All data about this user was cleared");
             return 0;
         }
+        std::println("There was no such user");
+        return 1;
+    }
+    std::println("Hello again(x{}),{}", db.UserIncrement(nameToFind), nameToFind);
+    return 0;
+}
 
-        std::ofstream fileEdit("database.txt", std::ios::trunc);
-        std::vector<std::string> arraySplitted=splitString(lineFound, ' ');
-       int num = std::stoi(arraySplitted[1]);
-       num++;
-       lines[lineNumber - 1] = arraySplitted[0] +" " + std::to_string(num);
-       for (const auto& line : lines) {
-           if (line != "")
-           {
-               fileEdit << line << std::endl;
-           }
-       }
-       std::cout << "Hello again(x"<<num <<"),"<< WordToFind << std::endl;
-       fileEdit.close();
+int main(int argc, char* argv[])
+{
+    Database db("database.txt");
+
+    if (argc < 2) {
+        std::println("Please provide a name to find.");
+        return 1;
+    }
+
+    std::string nameToFind;
+    nameToFind = argv[1];
+    const auto additionalCommand = std::string(argc > 2 ? argv[2] : "");
+
+    if (nameToFind == bread)
+    {
+        db.clearDatabase();
+        std::println("all data was cleared");
+        return 0;
+    }
+
+    if (!db.checkName(nameToFind)) {
+        return notFoundNameHandling(additionalCommand,nameToFind,db);
+    }
+    else {
+        return FoundNameHandling(additionalCommand,nameToFind,db);
     }
```

Comment:
Miinor: instead of
```
else {
        return FoundNameHandling(additionalCommand,nameToFind,db);
    }
return 0;
```
We can just write
`return FoundNameHandling(additionalCommand,nameToFind,db);`